qemu/tests/slof_device_tree.py

```diff
@@ -58,9 +59,10 @@ def check_nonexist_aliases(vm_session):
         uuid = vm.get_uuid()
         compare_dev_tree("system-id", uuid)
         compare_dev_tree("vm,uuid", uuid)
-        host_system_id = process.system_output(
-            "echo `cat /proc/device-tree/system-id`", shell=True).strip().decode()
-        compare_dev_tree("host-serial", host_system_id)
+        if int(utils_misc.get_qemu_version()['major']) < 4:
+            host_system_id = process.system_output(
+                "echo `cat /proc/device-tree/system-id`", shell=True).strip().decode()
```

Comment:
You can use process.getoutput() to get the decoded result.

Reply:
The function of process.system_output().decode() is not same with process.getoutput(), the test result is as following:
            host_system_id = process.system_output(
                "echo `cat /proc/device-tree/system-id`", shell=True).strip().decode()
            host_system_id_0 = process.system_output(
                "echo `cat /proc/device-tree/system-id`", shell=True).strip()
            host_system_id_1 = process.getoutput(
                "echo `cat /proc/device-tree/system-id`", shell=True).strip()
            host_system_id_2 = process.getoutput(
                "echo `cat /proc/device-tree/system-id`", shell=True).strip().splitlines()[-1]
            print("host_system_id is: %s" % host_system_id)
            print("host_system_id_0 is: %s" % host_system_id_0)
            print("host_system_id_1 is: %s" % host_system_id_1)
            print("host_system_id_2 is: %s" % host_system_id_2)
result:
host_system_id is: 130HJRA
host_system_id_0 is: b'130HJRA'
host_system_id_1 is: /bin/sh: warning: command substitution: ignored null byte in input
130HJRA
host_system_id_2 is: 130HJRA
Here, it just need the value "130HJRA", so, I think it does not need to update code.